src/app/parcels/add/AddParcelForm.tsx

```diff
@@ -0,0 +1,194 @@
+"use client";
+
+import React, { useState } from "react";
+import {
+    Errors,
+    FormErrors,
+    setError,
+    setField,
+    checkErrorOnSubmit,
+} from "@/components/Form/formFunctions";
+import {
+    CenterComponent,
+    StyledForm,
+    FormErrorText,
+    FormHeading,
+} from "@/components/Form/formStyling";
+
+import { useRouter } from "next/navigation";
+
+import VoucherNumberCard from "@/app/parcels/add/formSections/VoucherNumberCard";
+import PackingDateCard from "@/app/parcels/add/formSections/PackingDateCard";
+import TimeOfDayCard from "@/app/parcels/add/formSections/TimeofDayCard";
+import ShippingMethodCard from "@/app/parcels/add/formSections/ShippingMethodCard";
+import CollectionDateCard from "@/app/parcels/add/formSections/CollectionDateCard";
+import CollectionTimeCard from "@/app/parcels/add/formSections/CollectionTimeCard";
+import CollectionCentreCard from "@/app/parcels/add/formSections/CollectionCentreCard";
+import { insertParcel } from "@/app/parcels/add/databaseFunctions";
+import Button from "@mui/material/Button";
+
+interface AddParcelFields {
+    voucherNumber: string;
+    packingDate: Date | null;
+    timeOfDay: Date | null;
+    shippingMethod: string | null;
+    collectionDate: Date | null;
+    collectionTime: Date | null;
+    collectionCentre: string | null;
+}
+
+const withCollectionFormSections = [
+    VoucherNumberCard,
+    PackingDateCard,
+    TimeOfDayCard,
+    ShippingMethodCard,
+    CollectionDateCard,
+    CollectionTimeCard,
+    CollectionCentreCard,
+];
+
+const noCollectionFormSections = [
+    VoucherNumberCard,
+    PackingDateCard,
+    TimeOfDayCard,
+    ShippingMethodCard,
+];
+
+const initialFields: AddParcelFields = {
+    voucherNumber: "",
+    packingDate: null,
+    timeOfDay: null,
+    shippingMethod: "",
+    collectionDate: null,
+    collectionTime: null,
+    collectionCentre: null,
+};
+
+const initialFormErrors: FormErrors = {
+    voucherNumber: Errors.none,
+    packingDate: Errors.initial,
+    timeOfDay: Errors.initial,
+    shippingMethod: Errors.initial,
+    collectionDate: Errors.initial,
+    collectionTime: Errors.initial,
+    collectionCentre: Errors.initial,
+};
+
+const mergeDateAndTime = (date: Date, time: Date): Date => {
+    return new Date(
+        date.getFullYear(),
+        date.getMonth(),
+        date.getDate(),
+        time.getHours(),
+        time.getMinutes()
+    );
+};
+
+const AddParcelForm: React.FC<{ id: string }> = (props: { id: string }) => {
```

Comment:
- interface `id: string` (AddParcelFormProps?)
- `props` doesn't need to be used - `({id: string}) =>` suffices

Reply:
Done the interface, and changed to `{id}` (It thinks string is an unused variable when id: string is written)